game-core/src/main/java/games/strategy/engine/framework/startup/mc/ServerModel.java

```diff
@@ -602,7 +570,7 @@ public void allowRemoveConnections() {
 
     final Map<String, PlayerType> localPlayerMappings = new HashMap<>();
     // local player default = humans (for bots = weak ai)
-    final PlayerType defaultLocalType = headless
+    final PlayerType defaultLocalType = ui == null
```

Comment:
After looking at all these `ui == null` and `ui != null` expressions, it's a toss-up as to whether having a `headless` field is clearer.  Or maybe private helpers `isHeadless()` and possibly `isHeaded()`? :man_shrugging:

Reply:
This was supposed to be a temporary solution, but I realised that the UI object is actually getting passed around to another object that does the same: behaves as headless when UI is null, and performs actions on it when it isn't.
I'll have to start my journey somewhere else

Reply:
Actually I realized the 2 occurrences I found were either a null check, so effectively a boolean or only called by UI code anyways, so something that can easily be extracted.
I will follow-up in my next PR.

Reply:
EDIT 2:
Actually `ServerLauncher` is ugly AF. Even though only a boolean is supplied telling it whether or not it is headless, a UI component is passed when launching and set as "the current" ui object to refer to.
Who doesn't love legacy code 🤷‍♂️